Essentials/src/com/earth2me/essentials/services/ReserveEssentials.java

```diff
@@ -0,0 +1,1875 @@
+     * @param world The name of the {@link World} associated with the amount.
+     * @return True if the account has at least the specified amount of funds, otherwise false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncHasHoldings(String identifier, BigDecimal amount, String world) {
+        return CompletableFuture.supplyAsync(() -> hasHoldings(identifier, amount));
+    }
+
+    /**
+     * Used to determine if an account has at least an amount of funds.
+     * @param identifier The identifier of the account that is associated with this call.
+     * @param amount The amount you wish to use for this check.
+     * @param world The name of the {@link World} associated with the amount.
+     * @return True if the account has at least the specified amount of funds, otherwise false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncHasHoldings(UUID identifier, BigDecimal amount, String world) {
+        return CompletableFuture.supplyAsync(() -> hasHoldings(identifier, amount));
+    }
+
+    /**
+     * Used to determine if an account has at least an amount of funds.
+     * @param identifier The identifier of the account that is associated with this call.
+     * @param amount The amount you wish to use for this check.
+     * @param world The name of the {@link World} associated with the amount.
+     * @param currency The {@link Currency} associated with the balance.
+     * @return True if the account has at least the specified amount of funds, otherwise false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncHasHoldings(String identifier, BigDecimal amount, String world, String currency) {
+        return CompletableFuture.supplyAsync(() -> hasHoldings(identifier, amount));
+    }
+
+    /**
+     * Used to determine if an account has at least an amount of funds.
+     * @param identifier The identifier of the account that is associated with this call.
+     * @param amount The amount you wish to use for this check.
+     * @param world The name of the {@link World} associated with the amount.
+     * @param currency The {@link Currency} associated with the balance.
+     * @return True if the account has at least the specified amount of funds, otherwise false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncHasHoldings(UUID identifier, BigDecimal amount, String world, String currency) {
+        return CompletableFuture.supplyAsync(() -> hasHoldings(identifier, amount));
+    }
+
+    /**
+     * Used to set funds to an account.
+     * @param identifier The identifier of the account that is associated with this call.
+     * @param amount The amount you wish to set from this account.
+     * @return True if the funds were set for the account, otherwise false.
+     */
+    @Override
+    public boolean setHoldings(String identifier, BigDecimal amount) {
+        if (!hasAccount(identifier)) return false;
```

Comment:
So by this logic, an account that doesn't exist can have the minimum starting balance but can't have its holdings set? I understand this is a corner case, but cornercases I am out to find!